Sources/Apollo/PossiblyDeferred.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+typealias Thunk<Value> = () throws -> Value
```

Comment:
Yep, I should definitely add some documentation. Not sure about the name. I think `DeferredAction` doesn't really cover the meaning, because it is specifically a function that returns a value (but by that reasoning `Thunk` may not be specific enough either). We could also just leave out the type name and spell out the type? It's only used in one place right now.

Reply:
I feel like the typealias helps with readability due to the generic. Totally open to other ideas on naming, I just feel like `Thunk` has a specific, low-level meaning that I don't know we want to associate with this.

Reply:
Well, I suspect most people will associate it with the use of thunks in the Swift compiler, so that may be reason enough to pick another name, but it isn't low level in itself and has a general meaning that does seem appropriate to me:
>> Thunks are primarily used to delay a calculation until its result is needed, or to insert operations at the beginning or end of the other subroutine.